lib/redaction.js

```diff
@@ -77,7 +78,10 @@ function redaction (opts, serialize) {
   return [...Object.keys(shape), ...Object.getOwnPropertySymbols(shape)].reduce((o, k) => {
     // top level key:
     if (shape[k] === null) o[k] = topCensor
-    else o[k] = fastRedact({ paths: shape[k], censor, serialize, strict })
+    else {
+      const fr = fastRedact({ paths: shape[k], censor, serialize, strict })
+      o[k] = value => (typeof value === 'string' ? asString(value) : fr(value))
```

Comment:
I think this might be better fixed in fastRedact itself.

Reply:
Or maybe I misunderstood and it is a fast-redact thing.

Reply:
https://github.com/davidmarkclements/fast-redact is a little odd because it returns JSON text when an object is passed in, primitives otherwise (with `strict` set to `false`).  Perhaps it would be more consistent to always return JSON text.  When primitives were passed in, JSON.stringify() would be called for the return.

It would be a simple matter of changing the last lines of `lib/redactor.js` in `fast-redact`
```
function strictImpl (strict) {
  return strict === true ? `throw Error('fast-redact: primitives cannot be redacted')` : `return JSON.stringify(o)`    <--- used to be "return o"
}
```

but that breaks the existing unit test which checks that when a number is passed in, a number is returned.  The return could be written as:
`return typeof o === 'string' ? JSON.stringify(o) : o`
(basically the same as what is in this PR).   All tests would pass, but then the definition of `strict=false` would be a little awkward.  And the code doesn't use the cool optimized `asString()` function which pino has.

I think if you fixed `fast-redact` now, you'd be changing its documented behavior and so might break existing code in other packages which make use of it.

I suggest that a little caveat is added to the fast-readact markdown file warning of the `strict=false` treatment of string primitives.  And we work around this in the pino code.

Reply:
> I think if you fixed `fast-redact` now, you'd be changing its documented behavior and so might break existing code in other packages which make use of it.

I think we should release a new major version of fast-redact with the correct behavior, and then update pino. In this way other packages will not break unless they update to the new (fixed) version.

Would you mind sending a PR to fast-redact in this sense?